zds/tutorialv2/epub_utils.py

```diff
@@ -179,32 +187,44 @@ def copy_or_create_empty(src_path, dst_path, default_name):
             f.write('')
 
 
-def handle_images(relative_path):
-    def _(html_code):
-        soup_parser = BeautifulSoup(html_code, 'lxml')
-        for image in soup_parser.find_all('img'):
-            if not image.get('src', ''):
-                continue
-            image_url = image['src']
-            if image_url.startswith('http://') or image_url.startswith('https://'):
-                splitted = parse.urlsplit(image_url)
-                final_path = splitted.path
-            elif image_url.startswith(settings.MEDIA_URL):
-                final_path = Path(image_url).name
-            elif Path(image_url).is_absolute() and 'images' in image_url:
-                root = Path(image_url)
-                while root.name != 'images':
-                    root = root.parent
-                final_path = str(Path(image_url).relative_to(root))
-            else:
-                final_path = Path(image_url).name
-            image_path_in_ebook = relative_path + '/images/' + str(final_path).replace('%20', '_')
-            image['src'] = str(image_path_in_ebook)
-        ids = {}
-        for element in soup_parser.find_all(name=None, attrs={'id': (lambda s: True)}):
-            while element.get('id', None) and element['id'] in ids:
-                element['id'] += '-1'
-            if element.get('id', None):
-                ids[element['id']] = True
-        return soup_parser.prettify('utf-8').decode('utf-8')
-    return _
+class ImageHandling:
+    def __init__(self):
+        self.names = set()
+
+    def handle_images(self, relative_path):
+        def _(html_code):
```

Comment:
On peut renommer ça ? Pour le coup ça n'a aucune des deux significations habituelles (i18n ou variable non utilisée dans le parcour d'un itérable).

Reply:
oui je vais changer.